File: services/backend/models/job.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/v2/bson"
	"kayori.io/backend/data/mongorm"
)

type Job struct {
	mongorm.Model `bson:",inline"`
	Title         string          `bson:"title" json:"title"`
	Service       string          `bson:"service" json:"service"`
	Category      string          `bson:"category" json:"category"`
	Status        string          `bson:"status" json:"status"`
	Task          bson.D          `bson:"task" json:"task"`
	Projects      []bson.ObjectID `bson:"projects" json:"projects"`
	Schedule      *struct {
		Schedule bool      `bson:"schedule" json:"schedule"`
		Duration int       `bson:"duration" json:"duration"`
		Interval string    `bson:"interval" json:"interval"`
		LastRun  time.Time `bson:"last_run" json:"last_run"`
	} `bson:"schedule,omitempty" json:"schedule,omitempty"`
}

File: services/frontend/src/models/job.ts
export interface IJob {
    id: string;
    title: string;
    service: string;
    status: string;
    projects: string[];
    task: any;
}

export class Job implements IJob {
    id: string;
    title: string;
    service: string;
    status: string;
    projects: string[];
    task: any;

    constructor({ id, title, service, status, task, projects }: { id: string; title: string; service: string; status: string, task: any, projects: string[] }) {
        this.id = id;
        this.title = title;
        this.service = service;
        this.status = status;
        this.projects = projects
        this.task = task;
    }
}

